src/sentry/static/sentry/app/views/projectUserReports.jsx

```diff
@@ -52,7 +53,11 @@ const ProjectUserReports = createReactClass({
   },
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.location.search !== this.props.location.search) {
+    // Ignore changes to environment term since this is handled separately
+    const nextSearchTerm = omit(qs.parse(nextProps.location.search), 'environment');
+    const thisSearchTerm = omit(qs.parse(this.props.location.search), 'environment');
+
+    if (!isEqual(nextSearchTerm, thisSearchTerm)) {
```

Comment:
if they both change, this will still fetch twice, right?

Reply:
(its still an improvement so no worries)

Reply:
I guess so, but they wouldn't both change at the same time?